src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -136,24 +88,57 @@ class WorkspaceDashboardContent extends Component {
     }
   }
 
+  async save() {
+    const { refreshWorkspace, workspace: { workspace: { namespace, name } }, ajax: { Workspaces } } = this.props
+    const { editDescription: description } = this.state
+    try {
+      this.setState({ saving: true })
+      await Workspaces.workspace(namespace, name).shallowMergeNewAttributes({ description })
+      refreshWorkspace()
+    } catch (error) {
+      reportError('Error saving workspace', error)
+    } finally {
+      this.setState({ saving: false, editingWorkspace: false })
+    }
+  }
+
   render() {
-    const { workspace, refreshWorkspace, workspace: { accessLevel, workspace: { createdDate, lastModified, bucketName, attributes: { description } } } } = this.props
-    const { submissionsCount, storageCostEstimate, editingWorkspace } = this.state
+    const { workspace: { accessLevel, workspace: { createdDate, lastModified, bucketName, attributes: { description } } } } = this.props
+    const { submissionsCount, storageCostEstimate, editingWorkspace, editDescription = '', saving } = this.state
     const canWrite = Utils.canWrite(accessLevel)
     return div({ style: { flex: 1, display: 'flex', marginBottom: '-2rem' } }, [
       div({ style: styles.leftBox }, [
         div({ style: styles.header }, [
-          'About the project',
-          buttonSecondary({
+          div({ style: { display: 'inline-block', lineHeight: '2.25rem' } }, 'About the project'),
+          !editingWorkspace && buttonSecondary({
             style: { width: '2rem' },
             disabled: !canWrite,
             tooltip: !canWrite && 'You do not have permission to edit this workspace',
             onClick: () => this.setState({ editingWorkspace: true })
           }, [icon('edit')])
         ]),
-        description ?
-          h(ReactMarkdown, [description]) :
-          div({ style: { fontStyle: 'italic' } }, ['No description added'])
+        Utils.cond(
+          [
+            editingWorkspace, () => h(Fragment, [
+              h(SimpleMDE, {
+                options: {
+                  autofocus: true,
+                  initialValue: description,
+                  placeholder: 'Enter a description',
+                  status: false
+                },
+                value: editDescription,
+                onChange: editDescription => this.setState({ editDescription })
+              }),
+              div({ style: { display: 'flex', justifyContent: 'flex-end', margin: '1rem' } }, [
+                buttonSecondary({ onClick: () => this.setState({ editingWorkspace: false }) }, 'Cancel'),
+                buttonPrimary({ style: { marginLeft: '1rem' }, onClick: () => this.save() }, 'Save')
+              ]),
+              saving && spinnerOverlay
+            ])
+          ],
+          [description, h(ReactMarkdown, [description])],
```

Comment:
I don't like that we have two different markdown libraries. What about using SimpleMDE in both cases, but locking it in preview mode?